muni_portal/core/serializers.py

```diff
@@ -16,3 +19,38 @@ def to_representation(self, value):
         repr_value = super(SerializerMethodNestedSerializer, self).to_representation(value)
         if repr_value is not None:
             return self.serializer(repr_value, **self.serializer_kwargs).data
+
+
+class ImageSerializerField(Field):
+    """A custom serializer for image field."""
+
+    def to_representation(self, value):
+        return {
+            "url": value.file.url,
+            "width": value.width,
+            "height": value.height,
+            "alt": value.title
+        }
+
+
+class AdministratorPageSerializer(Field):
+    """A custom serializer for AdministratorPage."""
+
+    read_only = True
+    write_only = False
+
+    def to_representation(self, value):
+        if value.head_of_service and value.head_of_service.specific.profile_image:
```

Comment:
the head_of_service field is specific to the ServicePage, so it's weird that this serializer is called AdministratorPageSerializer.

Is this really what we have to do? So a field serializer has to have knowledge of all the specific pages where it might be used?

Reply:
This field is restricted to `AdministratorPage` page type `PageChooserPanel('head_of_service', 'core.AdministratorPage')` you can check this [here](https://github.com/OpenUpSA/muni-portal-backend/blob/7544c9d703863d9934287ec7d8586b72e6115064/muni_portal/core/models.py#L416).

So name `AdministratorPageSerializer` in this case tells us that we can serialize only this page. Field `head_of_service` we can pass as related name to serializer class, but for now, I think it is not necessary.